game/models.py

```diff
@@ -585,13 +586,44 @@ class Meta:
         )
 
 
-class CompetitionManager(models.Manager):
+class CompetitionMixin(object):
```

Comment:
Why don't we create BaseCompetitionManager, that contains the `get_visible_for_user` method, and create 3 managers for each competition state (past, ongoing, upcoming) and attach them to `past`, `ongoing` and `upcoming` attributes of the `Competition` model? This would allow us to write stuff like `Competition.ongoing.all()` or `Competition.upcoming.get_visible_for_user()`. What do you think?